src/Spatial/Euclidean/Line2D.cs

```diff
@@ -129,12 +135,53 @@ public Point2D ClosestPointTo(Point2D p, bool mustBeOnSegment)
 
             double t = (q - p).CrossProduct(s) / (r.CrossProduct(s));
 
-            if (double.IsPositiveInfinity(t) || double.IsNegativeInfinity(t))
+            return p + t * r;
+        }
+
+        /// <summary>
+        /// Compute the intersection between two lines if the angle between them is greater than a specified 
+        /// angle tolerance.
+        /// </summary>
+        /// <param name="other">The other line to compute the intersection with</param>
+        /// <returns>The point at the intersection of two lines, or null if the lines are parallel.</returns>
+        public Point2D? IntersectWith(Line2D other, Angle parallelTolerance)
+        {
+            if (this.IsParallelTo(other, parallelTolerance))
                 return null;
 
+            // http://stackoverflow.com/questions/563198/how-do-you-detect-where-two-line-segments-intersect
+            Point2D p = this.StartPoint;
+            Point2D q = other.StartPoint;
+            Vector2D r = this.StartPoint.VectorTo(this.EndPoint);
+            Vector2D s = other.StartPoint.VectorTo(other.EndPoint);
+
+            double t = (q - p).CrossProduct(s) / (r.CrossProduct(s));
+
             return p + t * r;
         }
 
+        /// <summary>
+        /// Checks to determine whether or not two lines are parallel to each other, using the dot product within 
+        /// the double precision specified in the MathNet.Numerics package.
+        /// </summary>
+        /// <param name="other">The other line to check this one against</param>
+        /// <returns>True if the lines are parallel, false if they are not</returns>
+        public bool IsParallelTo(Line2D other)
+        {
+            return this.Direction.IsParallelTo(other.Direction, Precision.DoublePrecision * 2);
```

Comment:
The test case failed with two identical vectors when it was just Precision.DoublePrecision.  In fact, the mismatch was exactly DoublePrecision * 2, so the comparison had to become a <= in order for it to return correct results.